### PR TITLE
Add grounding data (RAG) lab 2

### PR DESCRIPTION
**Pull request by jbusche**
Adding Lab 2 Rag Items

### CODE REVIEW

File: docs/lab-2/README.md [hunk 1]
```diff
@@ -4,13 +4,71 @@ description: Lab-2 of Prompt Engineering
...
 ---
 
-# Basics of Prompt Engineering
+# Learning about RAG (retrieval augmented generation)
 
 **Note:** The following images show actual results from watsonx.ai. The gray text is what we provided to the model. The blue highlighted text is how the model responded.
 
-### 1.0 LLM Foundations
-Before we jump into exploring the capabilities of Prompt Lab, we first need to lay a foundation for how Large Language Models (LLMs) work, and how we can tune the model and parameters to change their output. Gaining this understanding will make us more effective at prompt engineering.
+### 1.0 LLM RAG
```
Comment (rafvasq) [inline docs/lab-2/README.md]:
```suggestion
### 2.0 LLM RAG
```

File: docs/lab-2/README.md [hunk 2]
```diff
@@ -4,13 +4,71 @@ description: Lab-2 of Prompt Engineering
...
+This has multiple benefits.
+- Reduces hallucinations
+- Helps ensure the model has the most current, reliable facts
+- RAG also reduces the need to continiously retrain LLM models on new data, reducing time and cost
+
+![img.png](../images/lab2/2.1.0-groundingbutton.png)
 
 <img src="https://count.asgharlabs.io/count?p=/lab2_promptlab_page">
+
+### 1.1 How to add grounding data:
```
Comment (rafvasq) [inline docs/lab-2/README.md]:
```suggestion
### 2.1 How to add grounding data:
```

File: docs/lab-2/README.md [hunk 3]
```diff
@@ -4,13 +4,71 @@ description: Lab-2 of Prompt Engineering
...
-### 1.0 LLM Foundations
-Before we jump into exploring the capabilities of Prompt Lab, we first need to lay a foundation for how Large Language Models (LLMs) work, and how we can tune the model and parameters to change their output. Gaining this understanding will make us more effective at prompt engineering.
+### 1.0 LLM RAG
+RAG (retrieval-augmented generation) is the ability to retrieve facts from an external source to help ground LLMs and provide users detailed or updated information that otherwise might be missing in the LLM. It supplements the LLM's internal representation of information.
 
-![img.png](../images/lab1/1.1-free-form.png)
+This has multiple benefits.
+- Reduces hallucinations
+- Helps ensure the model has the most current, reliable facts
+- RAG also reduces the need to continiously retrain LLM models on new data, reducing time and cost
```
Comment (rafvasq) [inline docs/lab-2/README.md]:
```suggestion
- Reduces the need to continuously retrain LLM models on new data, reducing time and cost
```

File: docs/lab-2/README.md [hunk 4]
```diff
@@ -4,13 +4,71 @@ description: Lab-2 of Prompt Engineering
...
+Files can be added in PPTX, DOCX, PDF or TXT format with the following sizes:
+
+```
+PPTX: Up to 300 MB
+PDF : Up to 50 MB
+DOCX: Up to 10 MB
+TXT : Up to 5 MB
+```
+
+Click on Browse and add the `issue-36122.pdf` from your Downloads folder
```
Comment (rafvasq) [inline docs/lab-2/README.md]:
```suggestion
Click on "Browse" and add the file `issue-36122.pdf` you downloaded earlier.
```

File: docs/lab-2/README.md [hunk 5]
```diff
@@ -4,13 +4,71 @@ description: Lab-2 of Prompt Engineering
...
+```
+PPTX: Up to 300 MB
+PDF : Up to 50 MB
+DOCX: Up to 10 MB
+TXT : Up to 5 MB
+```
+
+Click on Browse and add the `issue-36122.pdf` from your Downloads folder 
+
+Note, the Name will auto-fill based on the name of the document you add
```
Comment (rafvasq) [inline docs/lab-2/README.md]:
```suggestion
Note, the "Name" field will auto-fill based on the name of the document you add.
```

File: docs/lab-2/README.md [hunk 6]
```diff
@@ -4,13 +4,71 @@ description: Lab-2 of Prompt Engineering
...
+
+![img.png](../images/lab2/2.1.1-addfile.png)
+
+Click "Create"
```
Comment (rafvasq) [inline docs/lab-2/README.md]:
```suggestion
Click "Create".
```

File: docs/lab-2/README.md [hunk 7]
```diff
@@ -4,13 +4,71 @@ description: Lab-2 of Prompt Engineering
...
+
+The chat interface then shows that it has the RAG (grounding data) appended as part of the chat:
+
+![img.png](../images/lab2/2.1.1-showchatwithgrounding.png)
+
+### 1.2 Using the grounding data
+Now you can ask questions and the grounding data will be used to help complete the answer and will indicate the pages it used for the source.
```
Comment (rafvasq) [inline docs/lab-2/README.md]:
```suggestion
Now, you can ask questions and the grounding data will be used to help complete the answer and will indicate the pages it used from the source.
```

File: docs/lab-2/README.md [hunk 8]
```diff
@@ -4,13 +4,71 @@ description: Lab-2 of Prompt Engineering
...
+
+For example, enter "Please summarize the issue" and you'll get something like this:
+
+![img.png](../images/lab2/2.1.2-usinggroundingdata.png)
+
+### 2.0 A note about the general chat feature
+Unlike the exercises where we were using Structured or Freeform mode, the Chat page has less customizable parameters to set for it, and the token limits are higher.  If you click on the "Edit System Prompt" you'll see some non-obvious instructions being provided to the LLM:
```
Comment (rafvasq) [inline docs/lab-2/README.md]:
```suggestion
Unlike exercises where we used Structured or Freeform mode, the Chat page has fewer customizable parameters and the token limits are higher. If you click on "Edit System Prompt", you'll see some non-obvious instructions being provided to the LLM:
```

File: docs/lab-2/README.md [hunk 9]
```diff
@@ -4,13 +4,71 @@ description: Lab-2 of Prompt Engineering
...
+
+```
+You are Granite Chat, an AI language model developed by IBM. You are a cautious assistant. You carefully follow instructions. You are helpful and harmless and you follow ethical guidelines and promote positive behavior. You are a AI language model designed to function as a specialized Retrieval Augmented Generation (RAG) assistant. When generating responses, prioritize correctness, i.e., ensure that your response is correct given the context and user query, and that it is grounded in the context. Furthermore, make sure that the response is supported by the given document or context. Always make sure that your response is relevant to the question. If an explanation is needed, first provide the explanation or reasoning, and then give the final answer. Avoid repeating information unless asked.
+```
+
+Clearly you can change these settings as needed, but this is a good starting point for a well-behaved AI assistant.
```
Comment (rafvasq) [inline docs/lab-2/README.md]:
```suggestion
You can always customize this prompt as needed, but it serves as a good starting point for a well-behaved AI assistant.
```

**Comment by jbusche** [comment]
Hey Rafael - all great suggestions, thank you!  I used the UI tool to commit the changes, but it appears that the DCO sign off doesn't work properly that way, at least for me..   let me see if I can amend it - let's see...

**Pull request merged**